src/lib/core/coio_file.c

```diff
@@ -506,7 +506,12 @@ coio_tempdir(char *path, size_t path_len)
 {
 	INIT_COEIO_FILE(eio);
 
-	const char *tmpdir = getenv("TMPDIR");
+	bool exceeded;
+	const char *tmpdir = getenv_safe("TMPDIR", path_len, &exceeded);
```

Comment:
Strictly speaking, it is not about your patch, but I found it confusing that we use words `len` and `size` interchangeably for buffers/strings. Personally I use `size` for a buffer size (including trailing `'\0'` if applicable) and `len` as a string length (without trailing `'\0'`, like in `strlen()`). I hope that this approach allows to avoid some confusion.

(Feel free to just close this discussion, it is more like a side thought.)

Reply:
No problem, I can rename `path_len` to `path_size` while I'm at it. Done.